### ISSUE TITLE
Upgrade to ginkgo v2.x

### ISSUE THREAD

**Issue by himanshu-kun** [gardener/machine-controller-manager#857]
**How to categorize this issue?**

/area dev-productivity
/kind enhancement
/priority 2

**What would you like to be added**:

Upgrade the IT of mcm-providers + framework part of MCM to use ginkgo v2.x
Pick up only after #853 as its higher prio + to avoid any pipeline IT problems

- [x] mcm (https://github.com/gardener/machine-controller-manager/pull/870)
- [x] provider-aws(https://github.com/gardener/machine-controller-manager-provider-aws/pull/151)
- [x] provider-az(https://github.com/gardener/machine-controller-manager-provider-azure/pull/127)
- [x] provider-gcp(https://github.com/gardener/machine-controller-manager-provider-gcp/pull/106)
- [x] provider-alicloud(https://github.com/gardener/machine-controller-manager-provider-alicloud/pull/69)
- [x] provider-openstack(https://github.com/gardener/machine-controller-manager-provider-openstack/pull/112)

**Why is this needed**:
To adapt to latest ginkgo changes. Since other projects have already migrated, its becoming painful to work on both simultaneously due to need to keep on changing the version.

**Comment by elankath** [comment]
Assigning this to myself, as I need the IT running with MCM master for another PR
